### ISSUE TITLE
no longer configures - x264_with_libav_normal_bitdepth

### ISSUE THREAD

**Issue by hydra3333** [rdp/ffmpeg-windows-build-helpers#226]
Hello.  The build script has worked successfully until today.   Now this error pops up.
The Ubuntu 16.10 VM and build script is unchanged from the last successful build a week or two ago.
- the Win32 and companion x65 folders were removed from the sandbox and the script was simply run again.
Note that the script was modified to always retrieve the latest from GIT, so would something have changed there to cause this ?

Suggestions on what to do would be very much welcomed.

```
configuring x264_with_libav_normal_bitdepth (/home/u/Desktop/ffmpeg-windows-build-helpers-withOpenCL-master/sandbox/win32/x264_with_libav_normal_bitdepth) as $ PKG_CONFIG_PATH=/home/u/Desktop/ffmpeg-windows-build-helpers-withOpenCL-master/sandbox/cross_compilers/mingw-w64-i686/i686-w64-mingw32/lib/pkgconfig PATH=/home/u/Desktop/ffmpeg-windows-build-helpers-withOpenCL-master/sandbox/cross_compilers/mingw-w64-i686/bin:$PATH ./configure --host=i686-w64-mingw32 --enable-static --cross-prefix=/home/u/Desktop/ffmpeg-windows-build-helpers-withOpenCL-master/sandbox/cross_compilers/mingw-w64-i686/bin/i686-w64-mingw32- --prefix=/home/u/Desktop/ffmpeg-windows-build-helpers-withOpenCL-master/sandbox/cross_compilers/mingw-w64-i686/i686-w64-mingw32 --enable-strip --extra-cflags=-mtune=generic --extra-cflags=-O3
Found no assembler
Minimum version is nasm-2.13
If you really want to compile without asm, configure with --disable-asm.
```

```
git checkout'ing origin/master
M	.gitignore
M	AUTHORS
M	COPYING
M	Makefile
M	common/aarch64/asm-offsets.c
M	common/aarch64/asm-offsets.h
M	common/aarch64/asm.S
M	common/aarch64/bitstream-a.S
M	common/aarch64/cabac-a.S
M	common/aarch64/dct-a.S
M	common/aarch64/dct.h
M	common/aarch64/deblock-a.S
M	common/aarch64/mc-a.S
M	common/aarch64/mc-c.c
M	common/aarch64/mc.h
M	common/aarch64/pixel-a.S
M	common/aarch64/pixel.h
M	common/aarch64/predict-a.S
M	common/aarch64/predict-c.c
M	common/aarch64/predict.h
M	common/aarch64/quant-a.S
M	common/aarch64/quant.h
M	common/arm/asm.S
M	common/arm/bitstream-a.S
M	common/arm/cpu-a.S
M	common/arm/dct-a.S
M	common/arm/dct.h
M	common/arm/deblock-a.S
M	common/arm/mc-a.S
M	common/arm/mc-c.c
M	common/arm/mc.h
M	common/arm/pixel-a.S
M	common/arm/pixel.h
M	common/arm/predict-a.S
M	common/arm/predict-c.c
M	common/arm/predict.h
M	common/arm/quant-a.S
M	common/arm/quant.h
M	common/bitstream.c
M	common/bitstream.h
M	common/cabac.c
M	common/cabac.h
M	common/common.c
M	common/common.h
M	common/cpu.c
M	common/cpu.h
M	common/dct.c
M	common/dct.h
M	common/deblock.c
M	common/frame.c
M	common/frame.h
M	common/macroblock.c
M	common/macroblock.h
M	common/mc.c
M	common/mc.h
M	common/mips/dct-c.c
M	common/mips/dct.h
M	common/mips/deblock-c.c
M	common/mips/macros.h
M	common/mips/mc-c.c
M	common/mips/mc.h
M	common/mips/pixel-c.c
M	common/mips/pixel.h
M	common/mips/predict-c.c
M	common/mips/predict.h
M	common/mips/quant-c.c
M	common/mips/quant.h
M	common/mvpred.c
M	common/opencl.c
M	common/opencl.h
M	common/opencl/bidir.cl
M	common/opencl/downscale.cl
M	common/opencl/intra.cl
M	common/opencl/motionsearch.cl
M	common/opencl/subpel.cl
M	common/opencl/weightp.cl
M	common/opencl/x264-cl.h
M	common/osdep.c
M	common/osdep.h
M	common/pixel.c
M	common/pixel.h
M	common/ppc/dct.c
M	common/ppc/dct.h
M	common/ppc/deblock.c
M	common/ppc/mc.c
M	common/ppc/mc.h
M	common/ppc/pixel.c
M	common/ppc/pixel.h
M	common/ppc/ppccommon.h
M	common/ppc/predict.c
M	common/ppc/predict.h
M	common/ppc/quant.c
M	common/ppc/quant.h
M	common/predict.c
M	common/predict.h
M	common/quant.c
M	common/quant.h
M	common/rectangle.c
M	common/rectangle.h
M	common/set.c
M	common/set.h
M	common/threadpool.c
M	common/threadpool.h
M	common/vlc.c
M	common/win32thread.c
M	common/win32thread.h
M	common/x86/bitstream-a.asm
M	common/x86/cabac-a.asm
M	common/x86/const-a.asm
M	common/x86/cpu-a.asm
M	common/x86/dct-32.asm
M	common/x86/dct-64.asm
M	common/x86/dct-a.asm
M	common/x86/dct.h
M	common/x86/deblock-a.asm
M	common/x86/mc-a.asm
M	common/x86/mc-a2.asm
M	common/x86/mc-c.c
M	common/x86/mc.h
M	common/x86/pixel-32.asm
M	common/x86/pixel-a.asm
M	common/x86/pixel.h
M	common/x86/predict-a.asm
M	common/x86/predict-c.c
M	common/x86/predict.h
M	common/x86/quant-a.asm
M	common/x86/quant.h
M	common/x86/sad-a.asm
M	common/x86/sad16-a.asm
M	common/x86/trellis-64.asm
M	common/x86/util.h
M	common/x86/x86inc.asm
M	common/x86/x86util.asm
M	doc/ratecontrol.txt
M	doc/regression_test.txt
M	doc/standards.txt
M	doc/threads.txt
M	doc/vui.txt
M	encoder/analyse.c
M	encoder/analyse.h
M	encoder/cabac.c
M	encoder/cavlc.c
M	encoder/encoder.c
M	encoder/lookahead.c
M	encoder/macroblock.c
M	encoder/macroblock.h
M	encoder/me.c
M	encoder/me.h
M	encoder/ratecontrol.c
M	encoder/ratecontrol.h
M	encoder/rdo.c
M	encoder/set.c
M	encoder/set.h
M	encoder/slicetype-cl.c
M	encoder/slicetype.c
M	example.c
M	extras/avisynth_c.h
M	extras/avxsynth_c.h
M	extras/cl.h
M	extras/cl_platform.h
M	extras/getopt.c
M	extras/getopt.h
M	extras/intel_dispatcher.h
M	extras/inttypes.h
M	extras/stdint.h
M	filters/filters.c
M	filters/filters.h
M	filters/video/cache.c
M	filters/video/crop.c
M	filters/video/depth.c
M	filters/video/fix_vfr_pts.c
M	filters/video/internal.c
M	filters/video/internal.h
M	filters/video/resize.c
M	filters/video/select_every.c
M	filters/video/source.c
M	filters/video/video.c
M	filters/video/video.h
M	input/avs.c
M	input/ffms.c
M	input/input.c
M	input/input.h
M	input/lavf.c
M	input/raw.c
M	input/thread.c
M	input/timecode.c
M	input/y4m.c
M	output/flv.c
M	output/flv_bytestream.c
M	output/flv_bytestream.h
M	output/matroska.c
M	output/matroska_ebml.c
M	output/matroska_ebml.h
M	output/mp4.c
M	output/mp4_lsmash.c
M	output/output.h
M	output/raw.c
M	tools/checkasm-a.asm
M	tools/checkasm-aarch64.S
M	tools/checkasm-arm.S
M	tools/checkasm.c
M	tools/digress/__init__.py
M	tools/digress/cli.py
M	tools/digress/comparers.py
M	tools/digress/constants.py
M	tools/digress/errors.py
M	tools/digress/scm/__init__.py
M	tools/digress/scm/dummy.py
M	tools/digress/scm/git.py
M	tools/digress/testing.py
M	tools/q_matrix_jvt.cfg
M	x264.c
M	x264.h
M	x264cli.h
M	x264dll.c
M	x264res.rc
HEAD is now at df79067... x86: Avoid self-relative expressions on macho64
Already up-to-date.
got no code changes, not forcing reconfigure for that...
configuring x264_with_libav_normal_bitdepth (/home/u/Desktop/ffmpeg-windows-build-helpers-withOpenCL-master/sandbox/win32/x264_with_libav_normal_bitdepth) as $ PKG_CONFIG_PATH=/home/u/Desktop/ffmpeg-windows-build-helpers-withOpenCL-master/sandbox/cross_compilers/mingw-w64-i686/i686-w64-mingw32/lib/pkgconfig PATH=/home/u/Desktop/ffmpeg-windows-build-helpers-withOpenCL-master/sandbox/cross_compilers/mingw-w64-i686/bin:$PATH ./configure --host=i686-w64-mingw32 --enable-static --cross-prefix=/home/u/Desktop/ffmpeg-windows-build-helpers-withOpenCL-master/sandbox/cross_compilers/mingw-w64-i686/bin/i686-w64-mingw32- --prefix=/home/u/Desktop/ffmpeg-windows-build-helpers-withOpenCL-master/sandbox/cross_compilers/mingw-w64-i686/i686-w64-mingw32 --enable-strip --extra-cflags=-mtune=generic --extra-cflags=-O3
Found no assembler
Minimum version is nasm-2.13
If you really want to compile without asm, configure with --disable-asm.
```

**Comment by kgp700** [comment]
Because this x264 commit (x86: Change assembler from yasm to nasm) - http://git.videolan.org/?p=x264/x264-sandbox.git;a=commit;h=d2b5f4873e2147452a723b61b14f030b2ee760a5

How to install nasm on MinGW?

**Comment by DeadSix27** [comment]
"On MingGW" ? What OS are you using, it has to be installed on that.

**Comment by hydra3333** [comment]
Ok thanks @kgp700 
@DeadSix27 "On MingGW" ? What OS are you using, it has to be installed on that."
I must be doing something wrong; Ubuntu 16.10 VM with nasm installed and latest x264 "master" from GIT and it still threw the issue above.
I wonder if it's something to do with a prefix in the configure and it's looking for nasm in mingw or something like that ?

**Comment by DeadSix27** [comment]
@hydra3333 type `nasm -v` 
Should be 2.13

**Comment by hydra3333** [comment]
ah. 
Ubuntu 16.10 tell me 2.12.01
time to build nasm from source I suppose.

**Comment by DeadSix27** [comment]
@hydra3333 Ye. not a big deal.

**Comment by hydra3333** [comment]
ok, will try something like this and fiddle until it works ...
```
download and unpack from http://www.nasm.us/pub/nasm/releasebuilds/2.13.01/nasm-2.13.01.tar.xz
cd nasm-2.13.01
./autogen.sh
./configure --prefix=$mingw_w64_prefix --exec_prefix=$mingw_w64_prefix --enable-sections --enable-lto
make #everything
make install --exec_prefix=$mingw_w64_prefix
```
in the mingw build script, at the end of function build_mingw_w64 ()

**Comment by DeadSix27** [comment]
@hydra3333 I did not install it to the mingw install, its not part of the toolchain imo, I installed it systemwide.

**Comment by hydra3333** [comment]
ok that makes sense.

**Comment by hydra3333** [comment]
Well, this didn't appear to crash
```
#------------------------------------------------------------------------------------------------
# 2017.05.26 x264 has a new dependency on nasm 2.13.1 or greater ... 
# before we do anything, build NASM if need be
set -x
if [[ ! -d "nasm-2.13.01" ]]; then
   echo "Downloading nasm 2.13.01"
   download_and_unpack_file "http://www.nasm.us/pub/nasm/releasebuilds/2.13.01/nasm-2.13.01.tar.xz"
   echo "Configuring nasm 2.13.01"
   cd nasm-2.13.01
      ./autogen.sh || exit 1
      ./configure --prefix=/usr --exec_prefix=/usr --enable-sections --enable-lto || exit 1 #
      echo "Make nasm 2.13.01"
      #make #everything 
      make || exit 1 #
      echo "Installing nasm 2.13.01"
      make install  || exit 1 #
   cd ..
   echo "Done Building and Installing nasm 2.13.01"
fi
set +x
#read -p "After nasm build, press Enter to continue"
#------------------------------------------------------------------------------------------------
```

**Comment by DeadSix27** [comment]
@hydra3333 that would require root, thus its unuseable in the ffmpeg script (unless you run that as root, which you shouldn't) either way I'd just manually install nasm or use a seperate script.

**Comment by kgpneok** [comment]
@hydra3333 Thanks:) resolved now

**Comment by hydra3333** [comment]
@DeadSix27 yes you are right of course.

**Comment by xiaowoniu2012** [comment]
on mac os just  "brew install nasm  " 
fix it.

**Comment by rdp** [comment]
If you want to add a configure time version check that would work (I think it already requires one to install nasm FWIW...) I'd be happy to merge that pull request. Thanks!

**Comment by VitalyZaikin** [comment]
You need to fix current version after install 2.13 

Woirking for me 

tar xzvf nasm-2.13.01.tar.gz 
cd nasm-2.13.01
./configure --prefix=/opt/nasm
make
sudo make install
export PATH=/opt/nasm/bin/:$PATH
